subsys/usb/device_next/usbd_core.c

```diff
@@ -260,13 +277,17 @@ int usbd_device_shutdown_core(struct usbd_context *const uds_ctx)
 
 static int usbd_pre_init(void)
 {
-	k_thread_create(&usbd_thread_data, usbd_stack,
-			K_KERNEL_STACK_SIZEOF(usbd_stack),
-			usbd_thread,
-			NULL, NULL, NULL,
-			K_PRIO_COOP(8), 0, K_NO_WAIT);
 
-	k_thread_name_set(&usbd_thread_data, "usbd");
+	STRUCT_SECTION_FOREACH(usbd_context, ctx) {
+		k_event_init(&ctx->events);
+		k_thread_create(ctx->thread_data, ctx->thread_stack,
+				ctx->stack_size,
+				usbd_thread,
+				ctx, NULL, NULL,
+				K_PRIO_COOP(8), 0, K_NO_WAIT);
```

Comment:
Why are there not one but many threads for USB here, and cooperative at that? Is there a time limit on how long these coop tasks are allowed to run before yielding?